app/scripts/controllers/createPersistentVolumeClaim.js

```diff
@@ -0,0 +1,115 @@
+ * @description
+ * # CreatePersistentVolumeClaimController
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('CreatePersistentVolumeClaimController', function ($filter, $routeParams, $scope, $window, ApplicationGenerator, DataService, Navigate, ProjectsService) {
+    $scope.alerts = {};
+    $scope.projectName = $routeParams.project;
+    $scope.serviceName = $routeParams.service;
+    $scope.accessModes="ReadWriteOnce";
+    $scope.pvc = {
+      name: $scope.serviceName || ""
+    };
+    $scope.claim = {};
+
+
+    $scope.breadcrumbs = [
+      {
+        title: $scope.projectName,
+        link: "project/" + $scope.projectName
+      },
+      {
+          title: "Storage",
+          link: "project/" + $scope.projectName + "/browse/storage"
+      },
+      {
+        title: "Request Storage"
+      }
+    ];
+
+    ProjectsService
+      .get($routeParams.project)
+      .then(_.spread(function(project, context) {
+        $scope.project = project;
+        // Update project breadcrumb with display name.
+        $scope.breadcrumbs[0].title = $filter('displayName')(project);
+        var labels = {},
+            orderByDisplayName = $filter('orderByDisplayName');
+
+        DataService.list("services", context, function(services) {
+//          console.log("Context:(list)", context);
+          $scope.services = orderByDisplayName(services.by("metadata.name"));
+          $scope.pvc.service = _.find($scope.services, function(service) {
+            return !$scope.serviceName || service.metadata.name === $scope.serviceName;
+          });
+          $scope.$watch('pvc.service', function() {
+            labels = angular.copy($scope.pvc.service.metadata.labels);
+          });
+        });
+
+        $scope.createPersistentVolumeClaim = function() {
+          if ($scope.createPersistentVolumeClaimForm.$valid) {
+            $scope.disableInputs = true;
+            var claimForm = $scope.createPersistentVolumeClaimForm.persistentVolumeClaimForm;
+            var claim = $scope._generatePersistentVolumeClaim({
+                                        labels: labels || {},
+                                        params: angular.extend({ include: true }, claimForm)
+                                      }, claimForm.name);
+
+            DataService.create('persistentvolumeclaims', null, claim, context)
+              .then(function() { // Success
+                // Return to the previous page
+                $window.history.back();
+              }, function(result) { // Failure
+                $scope.disableInputs = false;
+                $scope.alerts['create-persistent-volume-claim'] = {
+                  type: "error",
+                  message: "An error occurred requesting storage claim.",
+                  details: $filter('getErrorDetails')(result)
+                };
+              });
+          }
+        };
+
+        $scope._generatePersistentVolumeClaim = function(input){
```

Comment:
Doesn't need to be a scope function since it's not called from the view. I'd just use

```js
function _generatePersistentVolumeClaim() { 
  // ...
}
```